src/c++/library/http_client.cc

```diff
@@ -1898,47 +1899,64 @@ InferenceServerHttpClient::AsyncTransfer()
       return !this->ongoing_async_requests_.empty();
     });
 
-    curl_multi_perform(multi_handle_, &place_holder);
-    while ((msg = curl_multi_info_read(multi_handle_, &place_holder))) {
-      uintptr_t identifier = reinterpret_cast<uintptr_t>(msg->easy_handle);
-      auto itr = ongoing_async_requests_.find(identifier);
-      // This shouldn't happen
-      if (itr == ongoing_async_requests_.end()) {
-        std::cerr << "Unexpected error: received completed request that is not "
-                     "in the list of asynchronous requests"
-                  << std::endl;
-        curl_multi_remove_handle(multi_handle_, msg->easy_handle);
-        curl_easy_cleanup(msg->easy_handle);
-        continue;
-      }
-
-      long http_code = 400;
-      if (msg->data.result == CURLE_OK) {
-        curl_easy_getinfo(msg->easy_handle, CURLINFO_RESPONSE_CODE, &http_code);
-      } else if (msg->data.result == CURLE_OPERATION_TIMEDOUT) {
-        http_code = 499;
-      }
-
-      request_list.emplace_back(itr->second);
-      ongoing_async_requests_.erase(itr);
-      curl_multi_remove_handle(multi_handle_, msg->easy_handle);
-      curl_easy_cleanup(msg->easy_handle);
+    CURLMcode mc = curl_multi_perform(multi_handle_, &place_holder);
+    int numfds;
+    if (mc == CURLM_OK) {
+      // Wait for activity up to CLIENT_WAIT_TIME_MS
+      mc =
+          curl_multi_wait(multi_handle_, NULL, 0, CLIENT_WAIT_TIME_MS, &numfds);
```

Comment:
I don't understand why we require this value. Why not just use INT_MAX or `timeout_ms` value above?

Reply:
I thought it would be fine to do either. I added the `constexpr` to make sure we would return sooner (since it is acceptable to return early).